devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy

```diff
@@ -0,0 +1,392 @@
+	log.debug "productId:        ${cmd.productId}"
+	log.debug "productTypeId:    ${cmd.productTypeId}"
+	def msr = String.format("%04X-%04X-%04X", cmd.manufacturerId, cmd.productTypeId, cmd.productId)
+	updateDataValue("MSR", msr)
+	createEvent([descriptionText: "$device.displayName MSR: $msr", isStateChange: false])
+}
+
+private sendCommands(cmds, delay=1000) {
+	log.info "sendCommands called: cmds '${cmds}', delay '${delay}'"
+	sendHubCommand(cmds, delay)
+}
+
+private commands(commands, delay=200) {
+	log.info "commands called: commands '${commands}', delay '${delay}'"
+	delayBetween(commands.collect{ command(it) }, delay)
+}
+
+private secure(cmd) {
+	log.info "secure called"
+	if(zwaveInfo.zw.endsWith("s")) {
+		zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+	} else {
+		cmd.format()
+	}
+}
+
+private encap(cmd, endpoint = null) {
+	log.info "encap called"
+	if (endpoint) {
+		zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint:endpoint).encapsulate(cmd)
+	} else {
+		cmd
+	}
+}
+
+private secureEncap(cmd, endpoint = null) {
+	log.info "secureEncap called"
+	secure(encap(cmd, endpoint))
+}
+
+private changeSwitch(endpoint, value) {
+	log.info "changeSwitch called: vlaue: '${value}', endpoint: '${endpoint}'"
+	def result = []
+	if(endpoint) {
+		String childDni = "${device.deviceNetworkId}:$endpoint"
+		def child = childDevices.find { it.deviceNetworkId == childDni }
+		log.debug "changeSwitch: endpoint '${endpoint}', value: '${value}')"
+		result << child.sendEvent(name: "switch", value: value)
+	}
+	result
+}
+
+def childOnOff(deviceNetworkId, value) {
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) sendHubCommand onOffCmd(value, switchId)
+}
+
+def childRefresh(deviceNetworkId) {
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) sendHubCommand refresh(switchId)
+}
+
+def isDawonWallSmartSwitch3() {
+	return zwaveInfo.prod.equals("0063")
+}
+
+def isDawonWallSmartSwitch2() {
+	return zwaveInfo.prod.equals("0062")
+}
+
+def isDawonWallSmartSwitch1() {
+	return zwaveInfo.prod.equals("0061")
+}
+
+private onOffCmd(value, endpoint) {
+	log.info "onOffCmd called: val:${value}, ep:${endpoint}"
+	delayBetween([
+		secureEncap(zwave.basicV1.basicSet(value: value), endpoint),
+		"delay 500"
+	], 500)
+}
+
+def getSwitchId(deviceNetworkId) {
+	log.info "getSwitchId called: ${deviceNetworkId}"
+	def split = deviceNetworkId?.split(":")
+	return (split.length > 1) ? split[1] as Integer : null
+}
+
+private addChildSwitches(numberOfSwitches) {
+	log.info "addChildSwitches called: ${numberOfSwitches}"
+	for(def endpoint : 1..numberOfSwitches) {
+		try {
+			String childDni = "${device.deviceNetworkId}:$endpoint"
+			def componentLabel = "Dawon Smart Switch${endpoint}"
+			addChildDevice("Child Dawon Z-wave Wall Smart Switch Health", childDni, device.getHub().getId(), [
+				completedSetup: true,
+				label         : componentLabel,
+				isComponent   : false
```

Comment:
Is there a reason you've chosen not to add these device as components? You'd have to create new metadata, but right now all the children would show up as separate devices on the device list. I'm not sure how this device itself is physically configured, but I believe we generally want to treat physically integrated child devices as components.

Reply:
This part have intended by main component & child component.